packages/did-provider-ebsi/src/services/EbsiRPCService.ts

```diff
@@ -0,0 +1,157 @@
+ * @param {{ params: AddVerificationMethodParams[], id:number }} args
+ */
+export const addVerificationMethod = async (args: { params: AddVerificationMethodParams[]; id: number }): Promise<Response> => {
+  const { params, id } = args
+  const options = {
+    method: 'POST',
+    headers: new Headers({
+      Authorization: 'Bearer ' + token,
+    }),
+    body: JSON.stringify({
+      jsonrpc,
+      method: 'addVerificationMethod',
+      params,
+      id,
+    }),
+  }
+  return await (await fetch(baseUrl, options)).json()
+}
+
+/**
+ * Call to build an unsigned transaction to add a verification relationship. Requires an access token with "didr_write" scope.
+ * @param {{ params: AddVerificationMethodRelationshipParams[], id: number }} args
+ */
+export const addVerificationMethodRelationship = async (args: {
+  params: AddVerificationMethodRelationshipParams[]
+  id: number
+}): Promise<Response> => {
+  const { params, id } = args
+  const options = {
+    method: 'POST',
+    headers: new Headers({
+      Authorization: 'Bearer ' + token,
+    }),
+    body: JSON.stringify({
+      jsonrpc,
+      method: 'addVerificationMethodRelationship',
+      params,
+      id,
+    }),
+  }
+  return await (await fetch(baseUrl, options)).json()
+}
+
+/**
+ * Call to send a signed transaction to the blockchain. Requires an access token with "didr_invite" or "didr_write" scope.
+ * @param {{ params: SendSignedTransactionParams[], id: number }} args
+ */
+export const sendSignedTransaction = async (args: { params: SendSignedTransactionParams[]; id: number }): Promise<Response> => {
+  const { params, id } = args
+  const options = {
+    method: 'POST',
+    headers: new Headers({
+      Authorization: 'Bearer ' + token,
+    }),
+    body: JSON.stringify({
+      jsonrpc,
+      method: 'sendSignedTransaction',
+      params,
+      id,
+    }),
+  }
+  return await (await fetch(baseUrl, options)).json()
+}
+
+/**
+ * Gets the DID document corresponding to the DID.
+ * @param {GetDidDocumentParams} args
+ * @returns a did document
+ */
+export const getDidDocument = async (args: GetDidDocumentParams): Promise<DIDDocument> => {
+  const { did, validAt } = args
+  if (!did) {
+    throw new Error('did parameter is required')
+  }
+  const query = `?${validAt && `valid_at=${validAt}`}`
+  return await (await fetch(`https://api-pilot.ebsi.eu/did-registry/v5/identifiers/${did}${query}`)).json()
+}
+
+/**
+ * listDidDocuments - Returns a list of identifiers.
+ * @param {GetDidDocumentsParams} args
+ * @returns a list of identifiers
+ */
+export const listDidDocuments = async (args: GetDidDocumentsParams): Promise<GetDidDocumentsResponse> => {
+  const { offset, size, controller } = args
+  const query = `?${[offset && `page[after]=${offset}`, size && `page[size]=${size}`, controller && `controller=${controller}`]
+    .filter(Boolean)
+    .join('&')}`
+  return await (await fetch(`https://api-pilot.ebsi.eu/did-registry/v5/identifiers${query}`)).json()
```

Comment:
Please make sure that all the static links to EBSI are an optional parameter from a predefined set as well as string (string union). Then have a default value you can configure on the plugin for the EBSI environment. That way it will always use the default, and you can optionally provide a different env

Reply:
Added query url to getUrls method